### ISSUE TITLE
Issues with Safari (both mobile and desktop)

### ISSUE THREAD

**Issue by Dani-1-2** [ttalvitie/browservice#112]
Hello, I am using Safari 17.1 on Mac and Safari on iOS 10. I have noticed that when I enter a session, I have to refresh the webpage before the instance is loaded. Otherwise the webpage (including the interface) does not at all. When I look in the logs, it has started a new CEF browser, but nothing happens on the client until I refresh the page. This issue does not happen on Firefox. Any way to solve this?

**Comment by ttalvitie** [comment]
I am assuming that "entering session" means when you point the client browser to the Browservice HTTP address (and this should happen when the Browservice server is already running). I don't know what's happening here, but I guess this could be caused by the initialization sequence where we do some redirects to make the back/forward buttons work. When you experience the problem, what is the URL in the client browser address field? Does it have e.g. `prev/` or `next/` in the end?

**Comment by Dani-1-2** [comment]
> I am assuming that "entering session" means when you point the client browser to the Browservice HTTP address (and this should happen when the Browservice server is already running). I don't know what's happening here, but I guess this could be caused by the initialization sequence where we do some redirects to make the back/forward buttons work. When you experience the problem, what is the URL in the client browser address field? Does it have e.g. `prev/` or `next/` in the end?

It happens when I go to the normal url, as well as the /goto url.

**Comment by ttalvitie** [comment]
Yes, but what is the URL in the field after you have opened the Browservice page (after possible redirections)? I'm trying to see whether it gets stuck in some of the redirections.

**Comment by Dani-1-2** [comment]
> Yes, but what is the URL in the field after you have opened the Browservice page (after possible redirections)? I'm trying to see whether it gets stuck in some of the redirections.

It happens even before I enter anything in the field. It happens when I go to the browservice page without going to any address.

**Comment by Dani-1-2** [comment]
@ttalvitie 

https://github.com/ttalvitie/browservice/assets/149429107/dc070c69-3ae0-4209-a592-33b3d9e047ad

Here is a screen recording of the issue. The logs of browservice:
INFO @ retrojsvice.so src/window_manager.cpp:317 -- New window requested by user
INFO @ src/server.cpp:50 -- Got request for new window from vice plugin
INFO @ src/window.cpp:566 -- Creating window 3
INFO @ retrojsvice.so src/window_manager.cpp:328 -- Creating window 3
INFO @ src/window.cpp:190 -- CEF browser for window 3 created

**Comment by ttalvitie** [comment]
Ok, so it seems that the first redirect happens but then it gets stuck. This may be due to the client being somehow unhappy with the long chain of redirects we need to do to set up the client back/forward button forwarding. Could you try whether this happens with the newest version [Browservice v0.9.9.0](https://github.com/ttalvitie/browservice/releases/tag/v0.9.9.0) with the new command line option `--vice-opt-navigation-forwarding=disabled`? This disables the redirects that might be causing the issue.

**Comment by Dani-1-2** [comment]
@ttalvitie Seems to work now. I have another issue, but this time it is about running the server (completely separate from this), should I create another issue?
